newIDE/app/src/Export/LocalExporters/LocalFileSystem.js

```diff
@@ -31,7 +31,7 @@ export default {
     }
   },
   getTempDir: function() {
-    return os.tmpdir();
+    return path.join(os.tmpdir(), `GDTMP-${os.userInfo().uid}`);
```

Comment:
The [NodeJS documentation](https://nodejs.org/api/os.html#os_os_userinfo_options) about `uid` explain that return -1 for Windows, that mean this will be always -1 for all users session on Windows.

Reply:
The issue said the problem is only present on linux though, and on windows the temp directory is in app data so it is not shared and there are no such issues.

Reply:
Yes but the problem is only for linux

Reply:
Ok I think we can get a cleaner solution because I got this in my tmp folder.
![image](https://user-images.githubusercontent.com/1670670/116398461-1fc38e80-a828-11eb-8477-bcf844e6f721.png)

I suggest from current master :
```
| - tmp 
    | - GDGeneratedEventsFunctions
    | - GDTemporaries
    | - preview
```

To this one:

```
| - tmp 
    | - GDevelop
            | - 563B4512-6857-45C7-BD8F-E44A3386C4E8   (uid or -1 for windows users)
                    | - GDGeneratedEventsFunctions
                    | - GDTemporaries
                    | - preview
```

Reply:
Might not solve the problem on Linux if `tmp/GDevelop` is created and owned by a user?

Reply:
I mean, I agree for a `GDevelop-userId` single folder containing subfolders, but this can be another improvement unrelated to that fix. But I think for safety we would need to have the very first folder under `/tmp/` to be indexed by the user id, to avoid issue of "I don't have the permission to write in /tmp/GDevelop because it was originally created by another user"

Reply:
Contents in `tmp` are erased when the files are not used after a bunch of time, (e.g. `cleanmgr.exe` on Windows), so we can afford to have a tree structure of this type and at least there will be a list of users sessions that won't be listed directly in the OS's tmp folder.

Reply:
> Ok I think we can get a cleaner solution because I got this in my tmp folder.

To make it cleaner we can maybe make `LocalEventsFunctionCodeWriter.js` use the `getTempDir` from `LocalFileSystem.js` so that all use the `GDTMP-userid` folder?

Reply:
> Contents in tmp are erased when the files are not used after a bunch of time, (e.g. cleanmgr.exe on Windows), so we can afford to have a tree structure of this type and at least there will be a list of users sessions that won't be listed directly in the OS's tmp folder.

I think you are not getting the point of the PR, on linux, when creating a folder like this, it is in a global shared `/tmp` folder which is only cleared after exiting the application or restarting the system. If two users use GDevelop at the same time, and one makes for example a preview, then the other will not be able to write the preview files as the folder already exists and another user owns it. This PR adds the userid to the folder name so that each user has their own folder preventing one from not being able to write to another one due to permissions issues.

On Windows, this is irrelevant as there are no global temp folders, the temp folder is in appdata in the user directory: it is scoped to the user only.

Additionally, those are temporary files, an implementation detail. It is not made to be a clean and userfriendly directory structure but an efficient one.

Reply:
Yes GDevelop folder can be a blocking, your solution is fine.

```
| - tmp 
    | - GDevelop-563B4512-6857-45C7-BD8F-E44A3386C4E8
                    | - GDGeneratedEventsFunctions
                    | - GDTemporaries
                    | - preview
```

> To make it cleaner we can maybe make LocalEventsFunctionCodeWriter.js use the getTempDir from LocalFileSystem.js so that all use the GDTMP-userid folder?

Yes!

Reply:
It's no longer UID it's username I updated it.